### DISCUSSION TITLE
[solved] error when using opts (stable vs unstable branch)

### DISCUSSION THREAD

**Discussion by oliver-cfc** [nix-community/nixvim#1421] [General]
Firstly, I'm new to nix/flakes/home-manager so assume I don't know what I'm talking about.

 My setup is flakes/home-manager.  I have nixvim in the inputs and outputs in flake.nix and I import inputs.nixvim.homeManagerModules.nixvim in my home.nix.  Then I import nixvim.nix (a file I created)

All this is working until I put **_opts_** in nixvim.nix
The full nixvim.nix is:
```
{
  programs.nixvim = {
    enable = true;

    defaultEditor = true;
    viAlias = true;
    vimAlias = true;

    colorschemes.catppuccin.enable = true;

    plugins = {
      lualine.enable = true;
      bufferline.enable = true;
      treesitter.enable = true;
      which-key = {
        enable = true;
      };
    };

    opts = {};

  };
}

```
but this gives me an error when I try to 'home-manager switch --flake....'
```
error: The option `programs.nixvim.opts' does not exist. Definition values:
       - In `/nix/store/vkaad2fza7r8y418kjk9bcl97b2llg17-source/home-manager/pkgs/nixvim/nixvim.nix': { }
```
If I hash out the opts line, all it builds/switches as expected.

What am I missing?

**Comment by MattSturgeon** [comment]
The `opts` option used to be called `options`. If you are using a NixOS stable branch of nixvim _or_ your flake.lock hasn't been updated in a while then `opts` won't exist.

**Reply by oliver-cfc** [reply]
that was it exactly - stable branch.  Switched opts for options and everything built as expected.
Thanks!
